mods/tnt/init.lua

```diff
@@ -201,77 +229,56 @@ local function explode(pos, radius)
 			p.y = pos.y + y
 			p.z = pos.z + z
 			if cid ~= c_air then
-				destroy(drops, p, cid)
+				data[vi] = destroy(drops, p, cid, c_air, c_fire,
+					on_blast_queue, ignore_protection,
+					ignore_on_blast)
 			end
+
 		end
 		vi = vi + 1
 	end
 	end
 	end
 
+	vm:set_data(data)
+	vm:write_to_map()
+	vm:update_map()
+	vm:update_liquids()
+
+	for _, data in ipairs(on_blast_queue) do
+		local dist = math.max(1, vector.distance(data.pos, p))
+		local intensity = 1 / (dist * dist)
```

Comment:
According to `lua_api.txt` 1 is mid-range of regular TNT.  Either this should be changed or `lua_api.txt` should be updated.